### DISCUSSION TITLE
import recipes from text/word/google docs files

### DISCUSSION THREAD

**Discussion by tivo65** [mealie-recipes/mealie#2916] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/hay-kot/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

Been playing around with Mealie.  Love the program, thanks.  We have a bunch of recipes in google doc files, how would be the easiest way to import all those recipes.  

### Please provide a concise description of the feature that would resolve your issue.

mass import of text files/word docs/google doc files.  

### Please consider and list out some caveats or tradeoffs made in your design decision

the import may not be perfect, but maybe an easier way to get hundreds of files in

### Additional Information

- [ ] If this is accepted I'm willing to submit a PR to provide this feature
- [ ] If this is accepted I'm willing to help maintain this feature
- [ ] I'm willing to sponsor/pay a developer to do this work

**Comment by Kuchenpirat** [comment]
The problem with user generated text files is that there is no predictable structure to them.

So writing an parser for those files would be an enormous task.



The way I have handled my text files is that I have parsed them with an LLM (chatgpt in my case) into the recipe schema (https://schema.org/Recipe) and used this in combination with the mealie api to import those recipes.

Not a perfect solution, but certainly faster than to manually copy 50 recipes.

**Comment by tivo65** [comment]
very helpful, thanks.  chatgpt did a decent job converting the file into schema.org/recipe format.  i have not used the mealie api, what would be the easiest way to import this file into mealie.  what is the easist way to import just my one test json file?

**Reply by Kuchenpirat** [reply]
your mealie instance should have an `/docs` page with an documentation of the api and an simple way to execute api requests. 

This would be the best way to get startet.

**Reply by boc-the-git** [reply]
I haven't done it myself, but I think you want to first create the recipe: https://demo.mealie.io/docs#/Recipe%3A%20CRUD/create_one_api_recipes_post

Then do a patch request to populate the data:
https://demo.mealie.io/docs#/Recipe%3A%20CRUD/patch_one_api_recipes__slug__patch

Note I've linked to the demo site.. I expect you could do an initial test with that if easiest.

**Comment by michael-genson** [comment]
The newest nightly has the ability to import via image using OpenAI: https://github.com/mealie-recipes/mealie/pull/3974; it can also optionally translate the recipe for you (with varying success depending on which model you use)

While I'd like to get other import formats implemented (e.g. docx) at least for now you can convert pretty much anything into an image and feed it in that way
